sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/_tests_/e2e/student-scholastic-standings.institutions.controller.saveScholasticStanding.e2e-spec.ts

```diff
@@ -341,26 +552,277 @@ describe("StudentScholasticStandingsInstitutionsController(e2e)-saveScholasticSt
+      .expect({
+        statusCode: HttpStatus.BAD_REQUEST,
+        message: "Invalid submission.",
+        error: "Bad Request",
+      });
+  });
+
+  it("Should not create new scholastic standing Student withdrew from program for a part-time application when date of withdrawal is not between study start date and end date.", async () => {
+    // Arrange
+    mockFormioDryRun({
+      validDryRun: false,
+      studentScholasticStandingChangeType:
+        StudentScholasticStandingChangeType.StudentWithdrewFromProgram,
+      dateOfWithdrawal: "2023-04-01",
+    });
+    const application = await saveFakeApplication(
+      db.dataSource,
+      {
+        institutionLocation: collegeFLocation,
+      },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Institution token.
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/scholastic-standing/location/${collegeFLocation.id}/application/${application.id}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.BAD_REQUEST)
+      .expect({
+        statusCode: HttpStatus.BAD_REQUEST,
+        message: "Invalid submission.",
+        error: "Bad Request",
+      });
+  });
+
+  it("Should not create new scholastic standing Student completed program early for a part-time application when date of withdrawal is not between study start date and end date.", async () => {
+    // Arrange
+    mockFormioDryRun({
+      validDryRun: false,
+      studentScholasticStandingChangeType:
+        StudentScholasticStandingChangeType.StudentCompletedProgramEarly,
+      dateOfCompletion: "2023-04-01",
+    });
+    const application = await saveFakeApplication(
+      db.dataSource,
+      {
+        institutionLocation: collegeFLocation,
+      },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Institution token.
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/scholastic-standing/location/${collegeFLocation.id}/application/${application.id}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.BAD_REQUEST)
+      .expect({
+        statusCode: HttpStatus.BAD_REQUEST,
+        message: "Invalid submission.",
+        error: "Bad Request",
+      });
+  });
+
+  it("Should not create new scholastic standing Student did not complete program for a part-time application when date of change is greater than current date.", async () => {
```

Comment:
Same as in https://github.com/bcgov/SIMS/pull/3766/files#r1790978138

Reply:
https://github.com/bcgov/SIMS/pull/3766#discussion_r1792207469